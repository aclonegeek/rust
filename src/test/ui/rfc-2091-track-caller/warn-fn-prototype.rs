// run-pass

#![feature(track_caller)]
#![allow(dead_code)]

trait Trait {
    #[track_caller] //~ WARN: `#[track_caller]` is ignored on function prototypes
    fn foo(&self);
}

extern "Rust" {
    #[track_caller] //~ WARN: `#[track_caller]` is ignored on function prototypes
    fn bar();
}

fn main() {}
