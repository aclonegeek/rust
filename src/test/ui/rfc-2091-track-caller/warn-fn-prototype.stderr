warning: `#[track_caller]` is ignored on function prototypes
  --> $DIR/warn-fn-prototype.rs:12:5
   |
LL |     #[track_caller]
   |     ^^^^^^^^^^^^^^^
   |
   = note: `#[warn(unused_attributes)]` on by default

warning: `#[track_caller]` is ignored on function prototypes
  --> $DIR/warn-fn-prototype.rs:7:5
   |
LL |     #[track_caller]
   |     ^^^^^^^^^^^^^^^

